src/three/plugins/images/sources/GeoJSONImageSource.js

```diff
@@ -0,0 +1,423 @@
+import * as THREE from 'three';
+import { TiledImageSource } from './TiledImageSource.js';
+import { ProjectionScheme } from '../utils/ProjectionScheme.js';
+
+export class GeoJSONImageSource extends TiledImageSource {
+
+	constructor( {
+
+		geojson = null,
+		url = null, // URL or GeoJson object can be provided
+		tileDimension = 256,
+		bounds = [ - 180, - 90, 180, 90 ],
+		levels = 18,
+		projection = null, // 'EPSG:3857' or 'EPSG:4326' or null to auto- detect
+		pointRadius = 6,
+		strokeStyle = 'red',
+		fillStyle = 'rgba( 255,0,0,0.25 )'
+
+	} = {} ) {
+
+		super( );
+		this.geojson = geojson;
+		this.url = url;
+		this.tileDimension = tileDimension;
+		this.bounds = bounds;
+		this.levels = levels;
+		this.projection = projection;
+		this.pointRadius = pointRadius;
+		this.strokeStyle = strokeStyle;
+		this.fillStyle = fillStyle;
+
+	}
+
+	async init( ) {
+
+		// configure tiling to content bounds and levels
+		this.tiling.setProjection( new ProjectionScheme( this._detectProjection( ) ) );
+		this.tiling.setContentBounds( ...this.bounds );
+		for ( let i = 0; i < this.levels; i ++ ) {
+
+			const tilesX = 2 ** i;
+			const tilesY = 2 ** i;
+			this.tiling.setLevel( i, {
+				tilePixelWidth: this.tileDimension,
+				tilePixelHeight: this.tileDimension,
+				tileCountX: tilesX,
+				tileCountY: tilesY,
+			} );
+
+		}
+
+		// If a URL was provided and no geojson object yet, fetch it now (use fetchData so overlay can inject headers)
+		if ( ! this.geojson && this.url ) {
+
+			try {
+
+				const res = await this.fetchData( this.url );
+				// fetchData should return a Response-like object
+				// allow both Response and plain object (in case overlay fetch returns parsed JSON)
+				if ( res && typeof res.json === 'function' ) {
+
+					this.geojson = await res.json();
+
+				} else if ( res ) {
+
+					this.geojson = res;
+
+				}
+
+			} catch ( e ) {
+
+				console.warn( 'GeoJSONImageSource: failed to fetch geojson from url', this.url, e );
+				// continue, but without geojson no tiles will be produced
+
+			}
```

Comment:
I think we should just let this init function throw an error so it can be handled elsewhere if needed. This is what the other image sources do.